web-admin/src/features/projects/status/ProjectResources.svelte

```diff
@@ -10,41 +10,104 @@
   import { useQueryClient } from "@tanstack/svelte-query";
   import Button from "web-common/src/components/button/Button.svelte";
   import ProjectResourcesTable from "./ProjectResourcesTable.svelte";
+  import { ResourceKind } from "@rilldata/web-common/features/entity-management/resource-selectors";
+  import RefreshConfirmDialog from "./RefreshConfirmDialog.svelte";
+  import { onDestroy } from "svelte";
 
   const queryClient = useQueryClient();
   const createTrigger = createRuntimeServiceCreateTrigger();
 
-  let isReconciling = false;
+  let isRefreshConfirmDialogOpen = false;
+  let maxRefetchAttempts = 60; // 30 seconds maximum
+  let refetchAttempts = 0;
+  let pollInterval: ReturnType<typeof setInterval> | null = null;
+  let individualRefresh = false;
 
-  $: resources = createRuntimeServiceListResources(
+  $: allResources = createRuntimeServiceListResources(
     $runtime.instanceId,
-    // All resource "kinds"
     undefined,
     {
       query: {
         select: (data) => {
-          // Filter out the "ProjectParser" resource
+          // Exclude the "ProjectParser" resource and "RefreshTrigger" resource
           return data.resources.filter(
             (resource) =>
-              resource.meta.name.kind !== "rill.runtime.v1.ProjectParser",
+              resource.meta.name.kind !== ResourceKind.ProjectParser &&
+              resource.meta.name.kind !== ResourceKind.RefreshTrigger,
           );
         },
         refetchOnMount: true,
-        refetchInterval: isReconciling ? 500 : false,
+        keepPreviousData: true,
+        onError: () => {
+          stopPolling();
+        },
```

Comment:
1. `onError` will be deprecated in TanStack Query v5 ([see here](https://tkdodo.eu/blog/breaking-react-querys-api-on-purpose))
2. Rather than the imperative `startPolling()`, `stopPolling()`, have you considered using the built-in declarative `refetchInterval: () => number | false`?